pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -431,11 +443,20 @@ protected Array makeArray(int oid, String value) throws SQLException {
       return null;
     }
 
-    int oid = fields[i - 1].getOID();
+    int type = getSQLType(i);
+    if (type != Types.ARRAY) {
+      throw new PSQLException(
+          GT.tr("Cannot convert the column of type {0} to Array.",
+              JDBCType.valueOf(type).getName()),
+          PSQLState.DATA_TYPE_MISMATCH);
+    }
+
+    int oid = currentRow.fields()[i - 1].getOID();
     if (isBinary(i)) {
       return makeArray(oid, value);
     }
-    return makeArray(oid, castNonNull(getFixedString(i)));
+    boolean isStruct = getSQLType(i) == Types.STRUCT;
```

Comment:
didn't we just throw an exception above if the sql type is not Types.ARRAY? How could it ever be STRUCT here?